### DISCUSSION TITLE
How to pin Azure or another service as the default

### DISCUSSION THREAD

**Discussion by regstuff** [danny-avila/LibreChat#1376] [Q&A] [answered]
1. I have Azure, OpenAi & Bing configured on the app. I'd like to be able to make Azure the default for anyone using the app. But OpenAI seems to be the default always. I've tried rearranging the config sections in the .env file so Azure is on top, but no luck. Is there any way to do this?
2. I'm unable to change the APP title. Seems to be LibreChat no matter what I make it in the .env file. Custom footer works
3. Also unable to change the port from 3080.
Here's my .venv file (server section)
```
APP_TITLE=OfferingsChat
CUSTOM_FOOTER="Offerings Chat"
PORT=14004
```
4. Is there a way to stop the app from asking me to login every time I visit the page? I've set the expiry cookie to 7 days but no luck.
```
SESSION_EXPIRY=1000 * 60 * 60 * 24 * 7
REFRESH_TOKEN_EXPIRY=(1000 * 60 * 60 * 24) * 14
```
5. Wolfram plugin doesn't seem to work for me. When creating the App in Wolfram, should I choose LLM App or something else?
6. And finally, how do I share a preset? Saw this in the documentation: Share Custom Presets

Thank you

**Comment by danny-avila** [comment]
I can help with each of these items when I am home, thanks for your patience!

for now, I would say if you’re using Docker, you have to rebuild when the `.env` file is changed. 

For default order, check out dotenv.md, there is an ENDPOINTS variable and you can also set a default preset

**Comment by danny-avila** [comment, marked as answer]
> 1. I have Azure, OpenAi & Bing configured on the app. I'd like to be able to make Azure the default for anyone using the app. But OpenAI seems to be the default always. I've tried rearranging the config sections in the .env file so Azure is on top, but no luck. Is there any way to do this?

I can confirm that the following in the .env file makes this the "default" or first picked endpoint

![image](https://github.com/danny-avila/LibreChat/assets/110412045/4968db7f-7c9f-45b5-adab-88031d131cc6)

```bash
ENDPOINTS=azureOpenAI,openAI
```

Note that the app will use your last selected endpoint when creating a new conversation. So if you used OpenAI and hit new chat, it will have OpenAI selected with default conversation settings. To override this behavior, you need a preset and preset set as default.

I made a video to demonstrate: 

https://github.com/danny-avila/LibreChat/assets/110412045/bbde830f-18d9-4884-88e5-1bd8f7ac585d




> 2. I'm unable to change the APP title. Seems to be LibreChat no matter what I make it in the .env file. Custom footer works

This needs to be addressed due to the new UI. Will look into it.

> 4. Also unable to change the port from 3080.

This works for me

```
PORT=14004
````
![image](https://github.com/danny-avila/LibreChat/assets/110412045/e21b2688-ba62-42dc-86f8-fa23a1f030d3)

![image](https://github.com/danny-avila/LibreChat/assets/110412045/78ddc2d2-438a-44b6-8f90-1f2e7769d13c)

As I said before, if you’re using Docker, you have to rebuild when the .env file is changed. **Also if you're using docker, the port might have to be configured from the docker-compose file**

> 5. Is there a way to stop the app from asking me to login every time I visit the page? I've set the expiry cookie to 7 days but no luck.

TBD. Still need to investigate this but this isn't the case for me. By every time you visit is it on refresh? After a long period?

> 6. Wolfram plugin doesn't seem to work for me. When creating the App in Wolfram, should I choose LLM App or something else?

TBD. Will investigate but was working for me last time I used it

> 7. And finally, how do I share a preset? Saw this in the documentation: Share Custom Presets

Made a video to demonstrate:


https://github.com/danny-avila/LibreChat/assets/110412045/9eab8a5f-b92c-4e2a-8f20-94b72d62b2c3

**Comment by regstuff** [comment]
Thank you for the answers. Export presets works. The port issue was indeed something to be changed in the yml file. Also got Wolfram working. Not sure what I did wrong earlier?? But working fine now.

Regarding the repeated page login, it happens irrespective of the duration. Even refreshing the chat page triggers it.

**Reply by danny-avila** [reply]
You're welcome! Ah sounds familiar, could be a cookie issue: https://github.com/danny-avila/LibreChat/discussions/572#discussioncomment-7352331

The way around this in `production` mode is to use NGINX or something like Caddy for HTTPS if you are not outright accessing via `localhost`
